### DISCUSSION TITLE
Incorrect pins for the ESP32-S3-LCD-1.28 development board with GC9A01 Driver

### DISCUSSION THREAD

**Discussion by Sarooshki** [Bodmer/TFT_eSPI#3283] [Compatible displays and setup files]
I really would appreciate any help I could get because I cannot even figure out where the issue lies.
I am using this device (https://www.waveshare.com/esp32-s3-lcd-1.28.htm) that **has a built in ESP32-S2, this means I have no cables or connections other than the USB-C cable**
Tried run examples from the TFT library and other scripts that I obtained that use TFT scripts but nothing shows up on my screen.
I have used a demo script (unrelated to TFT) that does work, and it had the following pin configuration:

![Correct pin out](https://github.com/Bodmer/TFT_eSPI/assets/43258525/3133252a-0a1d-428c-b5d5-fb5ea6bd625b)

I went into the **"User_Setup.h"** file and the only definitions I have uncommented are the ones below as well as the "LOAD_FONT" ones:

#define USER_SETUP_INFO "User_Setup"
#define GC9A01_DRIVER
#define TFT_SDA_READ 

I went into **"User_Setup_Select.h"** and commented out line 27 (//#include <User_Setup.h>) and uncommented (#include <User_Setups/Setup46_GC9A01_ESP32.h>)

and then I went into **User_Setups/Setup46_GC9A01_ESP32.h** and I have the following code
![current pin out](https://github.com/Bodmer/TFT_eSPI/assets/43258525/7e87c71d-cf55-4c97-9a63-2dfbe4da04b6)

I also tried:
#define TFT_MOSI 11
#define TFT_SCLK 10
#define TFT_CS    9  
#define TFT_DC    8  
#define TFT_RST   12 

and I get nothing on my screen when I run any of the tests and diagnostics.
And my serial monitor does not show expected results.

And I also tried a different setup file called Setup302_Waveshare_ESP32S3_GC9A01.h because I thought it matched up more with the build in ESP32 rather than an external device. 
And this was my code, but I still do not get anything on my screen:

#define USER_SETUP_ID 302

#define GC9A01_DRIVER

#define TFT_MISO -1
#define TFT_MOSI 11
#define TFT_SCLK 10
#define TFT_CS    9  // Chip select control pin
#define TFT_DC    8  // Data Command control pin
#define TFT_RST   12  // Reset pin (could connect to RST pin)
#define TFT_BL 40
#define TFT_BACKLIGHT_ON HIGH

#define LOAD_GLCD   // Font 1. Original Adafruit 8 pixel font needs ~1820 bytes in FLASH
#define LOAD_FONT2  // Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
#define LOAD_FONT4  // Font 4. Medium 26 pixel high font, needs ~5848 bytes in FLASH, 96 characters
#define LOAD_FONT6  // Font 6. Large 48 pixel font, needs ~2666 bytes in FLASH, only characters 1234567890:-.apm
#define LOAD_FONT7  // Font 7. 7 segment 48 pixel font, needs ~2438 bytes in FLASH, only characters 1234567890:.
#define LOAD_FONT8  // Font 8. Large 75 pixel font needs ~3256 bytes in FLASH, only characters 1234567890:-.
#define LOAD_GFXFF  // FreeFonts. Include access to the 48 Adafruit_GFX free fonts FF1 to FF48 and custom fonts
#define SMOOTH_FONT

#define TFT_WIDTH 240
#define TFT_HEIGHT 240

**Comment by oshmykov-dev** [comment]
try this:
bl 2
dc 8
cs 9
sclk 10
mosi 11
miso 12
rst 14

**Reply by oshmykov-dev** [reply]
from here https://files.waveshare.com/wiki/ESP32-S3-Touch-LCD-1.28/ESP32-S3-Touch-LCD-1.28-Sch.pdf

**Reply by sroselli** [reply]
Hi, i have a problem with esp32-s3-dual-eye 1.28 notouch -whaveshare.
What'is a tutorial or correct configuration PIN's ? In this moment i bloked wirh display Black.

**Comment by bhagmeister** [comment]
the above numbers are the GPIOx numbers and not the dev board pin numbers. which is needed? i thought it would be the dev board pin numbers. also, be aware that the referenced schematic documents the RP2040-Touch-LCD-1.28 and not the ESP32S3….. (url name mismatch). try this link instead:   https://files.waveshare.com/wiki/ESP32-S3-LCD-1.28/Esp32-s3-lcd-.128-sch.pdf

**Comment by Sarooshki** [comment]
![Screenshot 2024-05-06 072246](https://github.com/Bodmer/TFT_eSPI/assets/43258525/b40a0813-593a-4309-8b61-dd1844b05490)

These are the correct pin outs and you use the GPIO pins. But somehow the connection is still not being made.

**Reply by oshmykov-dev** [reply]
![image](https://github.com/Bodmer/TFT_eSPI/assets/70232119/5f58d2a3-2ef0-4545-821d-96c66e0a3c1a)
yep I was using these GPIO pins in this project https://github.com/oshmykov-dev/esp-idf/tree/master/examples/peripherals/lcd/spi_lcd_touch

my board is a bit different though. This one https://www.amazon.com/dp/B0CM8L8HX2?ref=ppx_yo2ov_dt_b_product_details&th=1

**Reply by Sarooshki** [reply]
![Screenshot 2024-05-06 092344](https://github.com/Bodmer/TFT_eSPI/assets/43258525/7f620fbd-b420-49c6-b361-a34c36965ffd)

This is the one I have but even when I do match this in my Setup207_GC9A01.h file. Nothing shows up on the LCD.
So I am not even using User_setup.h. I just call out my Setup207_GC9A01.h in my User_Setup_Select.h.
Is there anything else that you do?
Because I try Read_User_Setup.ino in Test and Diagnostics and I can't even get proper output.
I really appreciate your time by the way, thank you.

**Reply by oshmykov-dev** [reply]
np at all. I can try the example on my board. Which one are you working with?

**Reply by Sarooshki** [reply]
In the TFT library by Bodmer, I try the Read_User_Setup.ino and I have tried this on different LCD I have and I get an output basically indicating what the pins I am using are to confirm that the set up is correct. But on my current LCD display. This picture is what I get from the output.
![User_setup_error](https://github.com/Bodmer/TFT_eSPI/assets/43258525/9affd563-7d30-48fc-8210-4fb89725d48a)

This is my code in the setup file

```
#define USER_SETUP_ID 207
#define USER_SETUP_INFO "ESP32-S3-Touch-LCD-1.28"

#ifdef ILI9341_DRIVER
#undef ILI9341_DRIVER
#endif

#define GC9A01_DRIVER

// For ESP32 Dev board (only tested with GC9A01 display)
// The hardware SPI can be mapped to any pins

#define TFT_WIDTH  240
#define TFT_HEIGHT 240

#define TFT_MISO -1
#define TFT_MOSI 11
#define TFT_SCLK 10
#define TFT_CS   9  // Chip select control pin
#define TFT_DC   8  // Data Command control pin
#define TFT_RST  12
#define TFT_BL	40
#define TFT_BACKLIGHT_ON HIGH

#define LOAD_GLCD   // Font 1. Original Adafruit 8 pixel font needs ~1820 bytes in FLASH
#define LOAD_FONT2  // Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
#define LOAD_FONT4  // Font 4. Medium 26 pixel high font, needs ~5848 bytes in FLASH, 96 characters
#define LOAD_FONT6  // Font 6. Large 48 pixel font, needs ~2666 bytes in FLASH, only characters 1234567890:-.apm
#define LOAD_FONT7  // Font 7. 7 segment 48 pixel font, needs ~2438 bytes in FLASH, only characters 1234567890:.
#define LOAD_FONT8  // Font 8. Large 75 pixel font needs ~3256 bytes in FLASH, only characters 1234567890:-.
#define LOAD_GFXFF  // FreeFonts. Include access to the 48 Adafruit_GFX free fonts FF1 to FF48 and custom fonts

#define SMOOTH_FONT

#define SPI_FREQUENCY  80000000
```

**Reply by Sarooshki** [reply]
One of my concerns that I cannot really check is that the User_Setup_ID in the file is an issue.. because waveshare created the setup file and they used 207. but when I look in the library I see that 207 is already there so I am not sure if that makes a difference. And I tried to just change to a random number like 400 so the IDs do not clash but that did not work.
So then I also just tried copying the configuration my from the Setup207_GC9A01.h file without the first two lines shown and I pasted to the bottom of the User_setup.h file and I uncommented the associated line on the User_Setup_Select.h file. But again, nothing.

and it is not like I did not use this library before. I used it with other waveshare screen and an external microcontroller. I just do not know why it is not working with this one.
```
#define USER_SETUP_ID 207
#define USER_SETUP_INFO "ESP32-S3-Touch-LCD-1.28"
```

**Reply by oshmykov-dev** [reply]
This is the header that appeared in Arduino/Libraries/TFT_eSPI_Setups

**Setup207_GC9A01.h**


```
#define USER_SETUP_ID 207
#define USER_SETUP_INFO "ESP32-S3-Touch-LCD-1.28"

#ifdef ILI9341_DRIVER
#undef ILI9341_DRIVER
#endif

#define GC9A01_DRIVER

// For ESP32 Dev board (only tested with GC9A01 display)
// The hardware SPI can be mapped to any pins

#define TFT_WIDTH  240
#define TFT_HEIGHT 240

#define TFT_MISO 12
#define TFT_MOSI 11
#define TFT_SCLK 10
#define TFT_CS   9  // Chip select control pin
#define TFT_DC   8  // Data Command control pin
#define TFT_RST  14
#define TFT_BL	2
#define TFT_BACKLIGHT_ON HIGH

#define LOAD_GLCD   // Font 1. Original Adafruit 8 pixel font needs ~1820 bytes in FLASH
#define LOAD_FONT2  // Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
#define LOAD_FONT4  // Font 4. Medium 26 pixel high font, needs ~5848 bytes in FLASH, 96 characters
#define LOAD_FONT6  // Font 6. Large 48 pixel font, needs ~2666 bytes in FLASH, only characters 1234567890:-.apm
#define LOAD_FONT7  // Font 7. 7 segment 48 pixel font, needs ~2438 bytes in FLASH, only characters 1234567890:.
#define LOAD_FONT8  // Font 8. Large 75 pixel font needs ~3256 bytes in FLASH, only characters 1234567890:-.
#define LOAD_GFXFF  // FreeFonts. Include access to the 48 Adafruit_GFX free fonts FF1 to FF48 and custom fonts

#define SMOOTH_FONT

#define SPI_FREQUENCY  80000000
```

**Reply by oshmykov-dev** [reply]
```
Rebooting...
�ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0xc (RTC_SW_CPU_RST),boot:0x38 (SPI_FAST_FLASH_BOOT)
Saved PC:0x4202846a
Warning: checksum mismatch between flashed and built applications. Checksum of built application is a4142f3fbb4671690ad4505f4211733cc34a16056e71218eafe078346a4192d5
0x4202846a: lv_obj_get_style_pad_column at C:/Users/oshmy/esp/spi_lcd_touch3/managed_components/lvgl__lvgl/src/core/lv_obj_style_gen.h:136
 (inlined by) lv_chart_get_point_pos_by_id at C:/Users/oshmy/esp/spi_lcd_touch3/managed_components/lvgl__lvgl/src/extra/widgets/chart/lv_chart.c:296

SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3808,len:0x4bc
load:0x403c9700,len:0xbd8
load:0x403cc700,len:0x2a0c
entry 0x403c98d0
Guru Meditation Error: Core  1 panic'ed (StoreProhibited). Exception was unhandled.

Core  1 register dump:
PC      : 0x420034eb  PS      : 0x00060830  A0      : 0x82003578  A1      : 0x3fcebbd0
0x420034eb: panic_print_registers at C:/Espressif/frameworks/esp-idf-v5.2.1/components/esp_system/port/arch/xtensa/panic_arch.c:89

A2      : 0x00000010  A3      : 0x00000000  A4      : 0x60004000  A5      : 0x0000000d
A6      : 0x000000ff  A7      : 0x00000000  A8      : 0x08000000  A9      : 0x3fcebba0
A10     : 0x3fc95308  A11     : 0x00000001  A12     : 0xffffffff  A13     : 0x00000004
A14     : 0x00000000  A15     : 0x3fc91d98  SAR     : 0x00000010  EXCCAUSE: 0x0000001d
EXCVADDR: 0x00000010  LBEG    : 0x42006a34  LEND    : 0x42006a97  LCOUNT  : 0x00000000
0x42006a34: get_adjusted_boot_time at C:/Espressif/frameworks/esp-idf-v5.2.1/components/newlib/time.c:92
0x42006a97: settimeofday at C:/Espressif/frameworks/esp-idf-v5.2.1/components/newlib/time.c:184



Backtrace: 0x420034e8:0x3fcebbd0 0x42003575:0x3fcebc00 0x420019ad:0x3fcebc20 0x4200780e:0x3fcebc50
0x420034e8: panic_print_registers at C:/Espressif/frameworks/esp-idf-v5.2.1/components/esp_system/port/arch/xtensa/panic_arch.c:89
0x42003575: panic_soc_fill_info at C:/Espressif/frameworks/esp-idf-v5.2.1/components/esp_system/port/arch/xtensa/panic_arch.c:426
0x420019ad: _stext at ??:?
0x4200780e: console_write at C:/Espressif/frameworks/esp-idf-v5.2.1/components/vfs/vfs_console.c:73





ELF file SHA256: 60ea35cc6b14f600
```

**Reply by oshmykov-dev** [reply]
It is also "207"

**Reply by LpaintQ** [reply]
same problem ,idk how to solve it

**Comment by jbglaw** [comment]
Hi!
I want to join here: I bought the same device (ESP32 with the round display but without touch: https://www.waveshare.com/wiki/ESP32-S3-LCD-1.28) and see the same issue with 2.5.43. Decoding the stack addresses, we're here:
```
Sketch: setup()

TFT: TFT_eSPI::init():692
 683   // Toggle RST low to reset
 684 #ifdef TFT_RST
 685   #if !defined(RP2040_PIO_INTERFACE)
 686     // Set to output once again in case MISO is used for TFT_RST
 687     if (TFT_RST >= 0) {
 688       pinMode(TFT_RST, OUTPUT);
 689     }
 690   #endif
 691   if (TFT_RST >= 0) {
 692*    writecommand(0x00); // Put SPI bus in known state for TFT with CS tied low
 693     digitalWrite(TFT_RST, HIGH);
 694     delay(5);
 695     digitalWrite(TFT_RST, LOW);
 696     delay(20);
 697     digitalWrite(TFT_RST, HIGH);
 698   }
 699   else writecommand(TFT_SWRST); // Software reset
 700 #else
 701   writecommand(TFT_SWRST); // Software reset
 702 #endif

TFT_eSPI::begin_tft_write(): 
  74 inline void TFT_eSPI::begin_tft_write(void){
  75   if (locked) {
  76     locked = false; // Flag to show SPI access now unlocked
  77 #if defined (SPI_HAS_TRANSACTION) && defined (SUPPORT_TRANSACTIONS) && !defined(TFT_PARALLEL_8_BIT) && !defined(RP2040_PIO_INTERFACE)
  78     spi.beginTransaction(SPISettings(SPI_FREQUENCY, MSBFIRST, TFT_SPI_MODE));
  79 #endif
  80     CS_L;
  81*    SET_BUS_WRITE_MODE;  // Some processors (e.g. ESP32) allow recycling the tx buffer when rx is not used
  82   }
  83 }
```
So it's hitting its head at the well due to a `StoreProhibited` within the code in the `SET_BUS_WRITE_MODE` macro.  Sounds like a setup issue with the SPI?

Tried to `#define TFT_RST -1` to get around the reset code, but it then hits the same way for the next step where it accesses the SPI.

**Comment by jbglaw** [comment]
Just found a hint on Reddid (https://www.reddit.com/r/arduino/comments/1bsl0l5/waveshare_esp32s3_128in_lcd_display_with_wifi/): You also need to `#define USE_HSPI_PORT`, so indeed this is a SPI setup issue. If that isn't already on the way, shall I suggest a config for the `User_Setups` directory?

**Reply by dustinbolton** [reply]
That reddit post solved the problem of getting it to work using the latest esp32 board firmware 3.0.4 for me. Before your fix I could only get the screen to work when using version 2.0.12 that is recommended in the Waveshare wiki docs. Thank you for sharing that fix!

**Comment by jbglaw** [comment]
In the repo, there is already a file that looks like if it was for this device. But it's missing the `#define USE_HSPI_PORT`, and the backlight setup is slightly different. Maybe that is for the touch version of this device? So I'd suggest to rename the current file (adding "touch" to its name) and place a second config.

**Comment by AA0100000101000001** [comment]
Hello, I'd like to report that I'm able to run the example in 2.5.34 with this custom setup
```C
#define USER_SETUP_ID 305
#define USER_SETUP_INFO "ESP32-S3-LCD-1.28"

#ifdef ILI9341_DRIVER
#undef ILI9341_DRIVER
#endif

#define GC9A01_DRIVER

#define TFT_WIDTH  240
#define TFT_HEIGHT 240

#define TFT_MOSI 11
#define TFT_SCLK 10
#define TFT_CS   9
#define TFT_DC   8
#define TFT_RST  12
#define TFT_BL   40
#define TFT_BACKLIGHT_ON HIGH

#define LOAD_GLCD
#define LOAD_FONT2
#define LOAD_FONT4
#define LOAD_FONT6 
#define LOAD_FONT7
#define LOAD_FONT8
#define LOAD_FONT8N
#define LOAD_GFXFF 

#define SMOOTH_FONT

#define SPI_FREQUENCY  80000000

```

Maybe something changed in the library? Are you using these settings? I had issues with different ones:

![image](https://github.com/Bodmer/TFT_eSPI/assets/144900857/b43903d1-1b29-4c42-bf62-582060e59e80)

**Comment by aghani-salforge** [comment]
Hi,
I have similar issue . But I figured out a solution.
First open User_Setup_Select.h & Comment Line 27 the add the following line & save the file
#include <User_Setups/Setup70h_ESP32_S3_GC9A01.h>  
![User_setup_select](https://github.com/Bodmer/TFT_eSPI/assets/162786255/4c5d5cef-7caf-4835-ab25-4e7e4b4e4330)

Now look for the User_Setups folder inside TFT_eSPI library folder. Open the User_Setups folder and look for the Setup70h_ESP32_S3_GC9A01.h file . Open it and adjust the pins as per your board  as mentioned below & save it.
![TFT_eSPI_folder](https://github.com/Bodmer/TFT_eSPI/assets/162786255/9feece1c-255e-4f12-b59d-50c1b13e49f4)
![setup_70h](https://github.com/Bodmer/TFT_eSPI/assets/162786255/cc4a7ac8-6c2b-4f2a-8577-cbd62b74a73a)

// Setup for the ESP32 S3 with GC9A01 display

#define USER_SETUP_ID 70
#define GC9A01_DRIVER

#define TFT_WIDTH  240

#define TFT_HEIGHT 240

 // Typical board default pins - change to match your board

#define TFT_CS   9 

#define TFT_MOSI 11 

#define TFT_SCLK 10 

#define TFT_MISO 13 

// Use pins in range 0-31

#define TFT_DC    8

#define TFT_RST   12

#define LOAD_GLCD
#define LOAD_FONT2
#define LOAD_FONT4
#define LOAD_FONT6
#define LOAD_FONT7
#define LOAD_FONT8
#define LOAD_GFXFF

#define SMOOTH_FONT

// FSPI port (SPI2) used unless following defined

#define USE_HSPI_PORT
#define SPI_FREQUENCY  80000000   // Maximum for ILI9341

**Reply by StuartMclean1502** [reply]
aghani-salforge you are a star. Have been struggling with that for a week, i'm practically bald now. But can confirm this does indeed work for me too.

**Reply by MaMaMarecal** [reply]
GREAT BRO !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! ADD THIS 
#define TFT_BL   40  // LED back-light
#define TFT_BACKLIGHT_ON HIGH

**Reply by DevDianDankie** [reply]
Thank you for sharing. I can go to sleep now. This works for me as well

**Comment by electricityforprogress** [comment]
Oh yes, amazing!  all of the examples and custom configs were not working for me until i added #define USE_HSPI_PORT  to the end of the User (or custom) setup file.  Now it is smooth sailing with pretty round pictures and LVGL!

**Comment by bionicbone** [comment]
I wish I'd found this thread before I spent so much time debugging LOL

I am using setup 302 (Setup302_Waveshare_ESP32S3_GC9A01) as the discription matches the board name, but as above the Key Part to stop the ESP32-S3 from crashing is to add this line into the user setup:
#define USE_HSPI_PORT
then change the pins to those above noted by @Sarooshki

**Comment by ofthedove** [comment]
I've been dealing with the same issue for a [similar board from SpotPear](https://spotpear.com/shop/ESP32-S3-1.28inch-Round-LCD-Display-TouchScreen.html)

Adding `USE_HSPI_PORT` got me working too!

I'm really curious as to why though. I've been trying to track it down. I also tracked it to `SET_BUS_WRITE_MODE`. Past there, it gets interesting.

That expands to `#define SET_BUS_WRITE_MODE *_spi_user = SPI_USR_MOSI` on line 125 of TFT_eSPI_ESP32_S3.h

The problem here is with `_spi_user`. It expands to: `(volatile uint32_t*)(SPI_USER_REG(SPI_PORT))` a few lines above.

With `USE_HSPI_PORT`, `SPI_PORT` expands to `3`. Without it, it expands to `FSPI` on the S3. However, over in `esp-hal-spi.h` in the esp32 arduino framework, `FSPI` expands to `0` if the the target is ESP32S3. 

Unfortunately, if you feed 0 into `SPI_USER_REG(i)`, which goes into `REG_SPI_BASE(i)`, it returns NULL. That `REG_SPI_BASE` calculation is done in `soc.h` in the arduino framework. It requires that the SPI enum be 2 or greater. 

~~I think the actual bug here might be in the Arduino framework. It's not clear to me why it's setting FSPI to 0 for `CONFIG_IDF_TARGET_ESP32S3`. The ESP32-S3 reference manual clearly shows FSPI is SPI2, not SPI0.~~

Edit: the framework only supports the use of SPI 3 for general purpose use. That's why FSPI is defined to zero, it should not be used. See the ESP IDF here: https://github.com/espressif/esp-idf/blob/9b3eda09741b5c7dae8f22fe60d193dcc3a7ec44/components/soc/esp32c3/include/soc/soc.h#L26C17-L26C106

**Comment by MaMaMarecal** [comment]
waveshare Guru Meditation Error: Core 1 panic'ed (StoreProhibited). Exception was unhandled
ESP32-S3-LCD-1.28
IF U ALSO HAVE THIS PROBLRM
COME HERE THIS THREAD

**Reply by ChR-iSz** [reply]
Has anyone the Partition Schema? My Code is to big...

**Reply by aghani-salforge** [reply]
Hi, 
Yes you can use partition scheme as 16M Flash (3 MB for App/ 9.9 MB for FATFS)

**Reply by ChR-iSz** [reply]
I try it 5 days with the partitions scheme. Then i see the waveshare has 8MB Flash instead 16MB (waveshare s3 r2, round Display). And NO PSRAM! Grrrr. Thanks waveshare for nothing!

**Reply by washa84** [reply]
Hello, i have the same issues. I tried different configurations, still have black screen, but finally without an error. My last result: 
2:19:28.403 -> ESP-ROM:esp32s3-20210327
02:19:28.403 -> Build:Mar 27 2021
02:19:28.403 -> rst:0x1 (POWERON),boot:0x38 (SPI_FAST_FLASH_BOOT)
02:19:28.403 -> SPIWP:0xee
02:19:28.403 -> mode:DIO, clock div:1
02:19:28.403 -> load:0x3fce2820,len:0x1188
02:19:28.403 -> load:0x403c8700,len:0x4
02:19:28.403 -> load:0x403c8704,len:0xbf0
02:19:28.403 -> load:0x403cb700,len:0x30e4
02:19:28.444 -> entry 0x403c88ac

I'm using this: 
#define USER_SETUP_ID 999
#define USER_SETUP_INFO "ESP32-S3-Touch-LCD-1.28"

#ifdef ILI9341_DRIVER
#undef ILI9341_DRIVER
#endif

#define GC9A01_DRIVER

// For ESP32 Dev board (only tested with GC9A01 display)
// The hardware SPI can be mapped to any pins

#define TFT_WIDTH  240
#define TFT_HEIGHT 240

//#define TFT_MISO 13
#define TFT_MOSI 11
#define TFT_SCLK 10
#define TFT_CS   9  // Chip select control pin
#define TFT_DC   8  // Data Command control pin
#define TFT_RST  12
#define TFT_BL	40
#define TFT_BACKLIGHT_ON HIGH

#define LOAD_GLCD   // Font 1. Original Adafruit 8 pixel font needs ~1820 bytes in FLASH
#define LOAD_FONT2  // Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
#define LOAD_FONT4  // Font 4. Medium 26 pixel high font, needs ~5848 bytes in FLASH, 96 characters
#define LOAD_FONT6  // Font 6. Large 48 pixel font, needs ~2666 bytes in FLASH, only characters 1234567890:-.apm
#define LOAD_FONT7  // Font 7. 7 segment 48 pixel font, needs ~2438 bytes in FLASH, only characters 1234567890:.
#define LOAD_FONT8  // Font 8. Large 75 pixel font needs ~3256 bytes in FLASH, only characters 1234567890:-.
#define LOAD_GFXFF  // FreeFonts. Include access to the 48 Adafruit_GFX free fonts FF1 to FF48 and custom fonts

#define SMOOTH_FONT

#define USE_HSPI_PORT
#define SPI_FREQUENCY  80000000

I tried with and without #define USE_HSPI_PORT, tried different TFT_MISO. Does anybody know, what else i can try? Thank you very much.

**Reply by bionicbone** [reply]
Looks like your back light is connected to pin 40?  Pin 40 should be a good one but assuming your LCD is 3.3v then move that wire directly to 3.3v just to check you are defo getting the BL signal, you could also try GND just in case is GND operated.
My only other observation is the SPI frequency is very high, does your LCD support that? Are the wires short enough to support that? I'd lower it to 20000000, at least until it's working.

**Reply by Basitadam** [reply]
Disconnect ESP pin 40 while driving BL manually. At least make it input

**Reply by washa84** [reply]
> Looks like your back light is connected to pin 40? Pin 40 should be a good one but assuming your LCD is 3.3v then move that wire directly to 3.3v just to check you are defo getting the BL signal, you could also try GND just in case is GND operated. My only other observation is the SPI frequency is very high, does your LCD support that? Are the wires short enough to support that? I'd lower it to 20000000, at least until it's working.

Hi, thanks for your reply. It's buit-in, display directly on the board, can't change the wire i think, or can i?

**Reply by washa84** [reply]
I found the solution. With every example directly from Waveshare i have no luck at all, a lot of errors during compilation, i can't upload even the basic demo test. 
Here is pinout: 
![image](https://github.com/user-attachments/assets/3d635232-7962-4605-8422-201250de7523)
And my settings: 
#define TFT_MISO 12
#define TFT_MOSI 11
#define TFT_SCLK 10
#define TFT_CS   9  // Chip select control pin
#define TFT_DC   8  // Data Command control pin
#define TFT_RST  14
#define TFT_BL	2
#define TFT_BACKLIGHT_ON HIGH

**Reply by bionicbone** [reply]
So you had back light set to pin 40 when the board requires pin 02, I can also see MISO has changed too

**Reply by washa84** [reply]
> So you had back light set to pin 40 when the board requires pin 02, I can also see MISO has changed too

Yes, now i can at least send a text to the display. But if i want to try ESP32-S3-LCD-1.28-Test, i still have an error: 

ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x1 (POWERON),boot:0x38 (SPI_FAST_FLASH_BOOT)
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce2820,len:0x1188
load:0x403c8700,len:0x4
load:0x403c8704,len:0xbf0
load:0x403cb700,len:0x30e4
entry 0x403c88ac

PSRAM is correctly initialized
GPIO Init successful!
Debug : Set image Rotate 0

drawing...

E (1628) i2c: i2c_master_cmd_begin(1535): Only allowed in master mode
E (1628) i2c: i2c_master_cmd_begin(1535): Only allowed in master mode
E (1629) i2c: i2c_master_cmd_begin(1535): Only allowed in master mode
QMI8658Register_WhoAmI = 255
E (1628) i2c: i2c_master_cmd_begin(1535): Only allowed in master mode
E (1628) i2c: i2c_master_cmd_begin(1535): Only allowed in master mode
E (1629) i2c: i2c_master_cmd_begin(1535): Only allowed in master mode
QMI8658_init fail
QMI8658_init

**Reply by bionicbone** [reply]
@washa84 can you send a link to that example code? So many examples over the years I can't find it

**Comment by SuperScuffer** [comment]
I'm using the Waveshare ESP32S3 touch version of this board (link below) and to get it to work I also needed to add  #define USE_HSPI_PORT to the end of the user setup file (in this case Setup302_Waveshare_ESP32S3_GC9A01.h).  

Thanks to everyone who contributed or I don't think I would have got this to work.

https://www.waveshare.com/wiki/ESP32-S3-Touch-LCD-1.28

**Comment by holgiw** [comment]
Hello, does tft.setRotation(x) work with values other than 0 with your GC9A01 driver?

**Reply by SuperScuffer** [reply]
Yes, this works fine.  The display rotates as expected (1, 2 etc).

**Comment by holgiw** [comment]
Okay yes true, my mistake, I meant the GC9D01 0.71" display.

**Comment by garethjpeers** [comment]
Hello I was hoping to get some help beacuse this board has been giving me problems and i have cannot get it to work so I was hoping that someone who got the screen to work could provide me with the user setup file and the code they got working on it. To clarify i am using the ESP32-S3-Touch-LCD-1.28 and the end goal of my project is to have a gps module display data on the screen while recording the data to and sd card. Thank you for any help provided.

**Reply by SuperScuffer** [reply]
My post above says which setup file worked for me and the line of code I added at the end.

**Comment by vishalsoniindia** [comment]
the issue is solve by defining 

#define USE_HSPI_PORT          // Force HSPI for esp32 v3.0 or above

https://github.com/vishalsoniindia/TFT_ESPI_not_working_with_esp32_v3.0.0_Above
